app/queries/get_unsuccessful_and_unsubmitted_candidates.rb

```diff
@@ -0,0 +1,13 @@
+class GetUnsuccessfulAndUnsubmittedCandidates
+  def self.call
+    Candidate
+    .joins(:application_forms)
+    .where(
+      application_forms: { recruitment_cycle_year: RecruitmentCycle.previous_year, id: ApplicationChoice.where(status: ApplicationStateChange::UNSUCCESSFUL_END_STATES).select(:application_form_id) },
```

Comment:
I think the conditions here are going to match on any application form that has at least one unsuccessful application choice. I am guessing that you really want application forms that have no successful or pending application choices?

Reply:
(just realised this was existing code from the renamed query).

Reply:
Oh good point! Will tweak the tests and see if i can get something working.

Reply:
@stevehook i think this is working now. Thanks for the feedback!